### ISSUE TITLE
Crash on long oids

### ISSUE THREAD

**Issue by xstaticxgpx** [xstaticxgpx/netsnmp-py3#3]
For example walk .1.3.6.1.2.1.4.24.7.1

**Comment by xstaticxgpx** [comment]
```
Python 3.4.3+ (default, Oct 10 2015, 09:15:38) 
[GCC 5.2.1 20151028] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import netsnmp
>>> with netsnmp.SNMPSession('archt01', 'public') as ss: [print(response) for response in ss.walk(['.1.3.6.1.2.1.4.24.7'])]
... 
('.1.3.6.1.2.1.4.24.7.1.7.1.4.0.0.0.0.0.2.0.0.1.4.192.168.1.1', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.7.1.4.0.0.0.0.0.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.7.1.4.192.168.1.0.24.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.7.1.4.192.168.1.1.32.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.7.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.128.3.0.0.7.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '1')
('.1.3.6.1.2.1.4.24.7.1.7.2.16.254.128.0.0.0.0.0.0.12.103.187.43.219.180.140.99.128.3.0.0.8.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '1')
('.1.3.6.1.2.1.4.24.7.1.7.2.16.255.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.8.3.0.0.9.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.8.1.4.0.0.0.0.0.2.0.0.1.4.192.168.1.1', 'INTEGER', '4')
('.1.3.6.1.2.1.4.24.7.1.8.1.4.0.0.0.0.0.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '3')
('.1.3.6.1.2.1.4.24.7.1.8.1.4.192.168.1.0.24.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '3')
('.1.3.6.1.2.1.4.24.7.1.8.1.4.192.168.1.1.32.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '3')
('.1.3.6.1.2.1.4.24.7.1.8.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.128.3.0.0.7.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '3')
('.1.3.6.1.2.1.4.24.7.1.8.2.16.254.128.0.0.0.0.0.0.12.103.187.43.219.180.140.99.128.3.0.0.8.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '3')
('.1.3.6.1.2.1.4.24.7.1.8.2.16.255.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.8.3.0.0.9.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '3')
('.1.3.6.1.2.1.4.24.7.1.9.1.4.0.0.0.0.0.2.0.0.1.4.192.168.1.1', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.9.1.4.0.0.0.0.0.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.9.1.4.192.168.1.0.24.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.9.1.4.192.168.1.1.32.3.0.0.2.1.4.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.9.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.128.3.0.0.7.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.9.2.16.254.128.0.0.0.0.0.0.12.103.187.43.219.180.140.99.128.3.0.0.8.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.9.2.16.255.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.8.3.0.0.9.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'INTEGER', '2')
('.1.3.6.1.2.1.4.24.7.1.10.1.4.0.0.0.0.0.2.0.0.1.4.192.168.1.1', 'Gauge32', '0')
('.1.3.6.1.2.1.4.24.7.1.10.1.4.0.0.0.0.0.3.0.0.2.1.4.0.0.0.0', 'Gauge32', '0')
('.1.3.6.1.2.1.4.24.7.1.10.1.4.192.168.1.0.24.3.0.0.2.1.4.0.0.0.0', 'Gauge32', '0')
('.1.3.6.1.2.1.4.24.7.1.10.1.4.192.168.1.1.32.3.0.0.2.1.4.0.0.0.0', 'Gauge32', '0')
('.1.3.6.1.2.1.4.24.7.1.10.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.128.3.0.0.7.2.16.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0', 'Gauge32', '0')
*** Error in `python3': realloc(): invalid old size: 0x00007ffda314ab10 ***
```

**Comment by xstaticxgpx** [comment]
See traceback at https://gist.github.com/xstaticxgpx/8b7039d863566eda5315

**Comment by xstaticxgpx** [comment]
(tentatively) fixed in a5aeb3e0cf53126723b3b3c450122ddcf0953303
